src/main/java/org/truffleruby/language/yield/CallBlockNode.java

```diff
@@ -74,7 +74,7 @@ protected Object callBlockUncached(
                 declarationContext,
                 block.frameOnStackMarker,
                 self,
-                (RubyProc) blockArgument,
+                blockArgument,
                 arguments);
```

Comment:
The comment above in this file `// blockArgument is typed as Object below because it must accept "null".` can be removed, because it no longer applies (it still needs to be Object, but not because of `null`).

Reply:
Good catch, thanks.